securedrop/tests/common.py

```diff
@@ -1,120 +1,272 @@
+# -*- coding: utf-8 -*-
+
+from functools import wraps
+import gnupg
+import mock
 import os
 import shutil
-import uuid
 import subprocess
-
-import gnupg
+import time
 
 # Set environment variable so config.py uses a test environment
 os.environ['SECUREDROP_ENV'] = 'test'
+
 import config
-from db import init_db, db_session, Journalist, Reply, Source, Submission
+from db import db_session, init_db, Journalist, Reply, Source, Submission
 import crypto_util
+import store
 
-# TODO: the PID file for the redis worker is hard-coded below.
-# Ideally this constant would be provided by a test harness.
-# It has been intentionally omitted from `config.py.example`
-# in order to isolate the test vars from prod vars.
-# When refactoring the test suite, the TEST_WORKER_PIDFILE
-# TEST_WORKER_PIDFILE is also hard-coded in `manage.py`.
-TEST_WORKER_PIDFILE = "/tmp/securedrop_test_worker.pid"
 
+class TestJournalist:
+    """Test fixtures for working with :class:`db.Journalist`.
+    """
+    @staticmethod
+    def init_journalist(is_admin=False):
+        """Initialize a journalist into the database. Return their
+        :class:`db.Journalist` object and password string.
 
-def clean_root():
-    shutil.rmtree(config.SECUREDROP_DATA_ROOT)
+        :param bool is_admin: Whether the user is an admin.
 
+        :returns: A 2-tuple. The first entry, the :class:`db.Journalist`
+                  initialized. The second, their password string.
+        """
+        username = crypto_util.genrandomid()
+        user_pw = crypto_util.genrandomid()
+        user = Journalist(username, user_pw, is_admin)
+        db_session.add(user)
+        db_session.commit()
+        return user, user_pw
 
-def create_directories():
-    # Create directories for the file store and the GPG keyring
-    for d in (config.SECUREDROP_DATA_ROOT, config.STORE_DIR,
-              config.GPG_KEY_DIR, config.TEMP_DIR):
-        if not os.path.isdir(d):
-            os.mkdir(d)
+    @staticmethod
+    def init_admin():
+        return TestJournalist.init_journalist(True)
+
+    @staticmethod
+    def mock_verify_token(self):
```

Comment:
Nvm--answered too quick. It's supposed to be there. Look at how it's used in `test_journalist` and `test_unit_journalist`. Should probably rename the variable to `app`.